mnefun/_report.py

```diff
@@ -40,6 +40,7 @@
 @contextmanager
 def report_context():
     """Create a context for making plt and mlab figures."""
+    mne.viz.set_browser_backend('matplotlib')
```

Comment:
you should use the https://mne.tools/dev/generated/mne.viz.use_browser_backend.html context manager so that it is only set temporarily

Reply:
So, I tried use_browser_backend first, but that failed for me (I still got a qt instance). :) I didn't dig in to why it wasn't working -- thoughts?

Reply:
Did you do it in a "with" clause? It won't work if it's not done that way. Even though it's within another context manager, create a second context within it for the browser backend.

Reply:
Specifying my above comment: I think I had trouble because I don't have experience nesting context managers inside of other context managers. I can see it's possible because in the `report_context()` function, you do this: 

```        
try:
    with plt.style.context(style):
        yield
except Exception:
    matplotlib.use(old_backend, force=True)
    plt.interactive(is_interactive)
    raise```

to use the plt style context. But what's the functional and proper way to do that embedding with `mne.viz.use_browser_backend` ?  Then I was having doubts about how many levels of context nesting make sense! But I do see the advisability of only setting the backend temporarily.

Reply:
You can do as many levels you want
```
with a:
    with b:
        ...
```
Or if you will want them both active the whole `.. ` of code (nothing with just `a`) then you can instead do
```
with a, b:
    ...
```

Reply:
Here's one example... 
![Screenshot from 2023-04-01 21-18-18](https://user-images.githubusercontent.com/38704848/229331303-59b82411-917b-414b-959b-28dac8d5c3f5.png)

No plot pops up at all. ??? Am I missing something obvious? lol

Reply:
@drammock I tried this also (which, I think, is what you described)

![Screenshot from 2023-04-01 21-35-45](https://user-images.githubusercontent.com/38704848/229331550-a363bafc-9c3c-4144-b93c-7b1e052d0ada.png)

..which gives me the following, without any plot at all. 

![Screenshot from 2023-04-01 21-36-07](https://user-images.githubusercontent.com/38704848/229331560-804f5aba-8e30-4193-a77a-8e1ece4eb09c.png)

Reply:
P.S. This is not urgent, fyi. :smile:  (I just needed to get reports working *enough* that I could run them on the subjects reprocessed with the new calibration files, and that got done Friday night.) Sorry to talk at you on the weekend -- please ignore till convenient!

Reply:
Without looking at the code... I think in report_context we probably set the backend to Agg, which is a non-GUI backend. So nothing popping up is the correct/desired behavior here!

Reply:
Alrighty, friends, I propose the following fix. Basically, I think my issue is Pycharm, which tries to do some backend handling stuff when importing matplotlib. If I call `with mne.viz.use_backend_browser('matplotlib'):`  *after* importing matplotlib and forcing it to use `Agg` in the `report_context()` function, then everything works.

Reply:
@larsoner (Sorry, I didn't see your comment earlier). `Agg` shouldn't cause your console to freeze such that you have to force quit it though, right??  :laughing:  I ran the above code from interactive python in a terminal earlier today, and all of the above code options worked perfectly from the terminal, so that makes me feel pretty confident that Pycharm is what's causing my problems!!

Reply:
🌩️#️⃣⁉️pycharm💩

Reply:
@drammock  You know, you were the one who told me back in the day that checking your code in the terminal is a good troubleshooting step. Thanks for the good advice!!! :laughing: